packages/client/src/object/Attachment.ts

```diff
@@ -23,3 +26,17 @@ export class Attachment implements IAttachment {
 export function isAttachment(o: any): o is Attachment {
   return o instanceof Attachment;
 }
+
+export function isAttachmentUpload(o: any): o is IAttachmentUpload {
+  return o instanceof Blob && "name" in o;
+}
+
+export function createAttachmentUpload(
+  data: Blob,
+  name: string,
+): IAttachmentUpload {
+  const attachmentUpload = Object.create(data);
+  Object.assign(attachmentUpload, { name });
+
+  return attachmentUpload as IAttachmentUpload;
```

Comment:
This could probably just be `return Object.create(data, { name: {value: name} });` or maybe slightly more friendly: `return Object.create(data, { name: {value: name, enumerable: true} });`

The enumerable isn't needed for `in`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties